index.js

```diff
@@ -1,38 +1,34 @@
 'use strict';
 
+const { IncomingMessage, ServerResponse } = require('http');
 const Layer = require('express/lib/router/layer');
-const Router = require('express/lib/router');
+const { Router } = require('express');
 
-const last = (arr = []) => arr[arr.length - 1];
-const noop = Function.prototype;
+const isError = (arg) => arg instanceof Error;
+const isRequest = (arg) => arg instanceof IncomingMessage;
+const isResponse = (arg) => arg instanceof ServerResponse;
+const noop = () => {};
 
-function copyFnProps(oldFn, newFn) {
-  Object.keys(oldFn).forEach((key) => {
-    newFn[key] = oldFn[key];
-  });
-  return newFn;
+function copyProps(source, dest) {
+  return Object.keys(source).reduce((acc, key) => {
+    const value = source[key];
+    return Object.assign(acc, { [key]: value });
+  }, dest);
```

Comment:
Can you expand on the reason for this refactor?

Reply:
I thought `reduce` serves better purpose conveying message to the reader because source object is _reduced_ (Yeah it is mapped but every mapper is in fact a reducer) to destination one. Plus `reduce` has return value compared to `forEach` which I thought is neat and leveraged here.